src/core/Cargo.toml

```diff
@@ -7,10 +7,11 @@ repository = "https://github.com/sourmash-bio/sourmash"
 keywords = ["minhash", "bioinformatics"]
 categories = ["science", "algorithms", "data-structures"]
 license = "BSD-3-Clause"
-edition = "2018"
+edition = "2021"
```

Comment:
Since we are on `1.56.1` now, also bump the `edition`.

This also fix the current beta lint failure on `latest`.